archetypes/archetype-lambda/README.md

```diff
@@ -0,0 +1,65 @@
+# Maven Archetype for lambda function using AWS SDK for Java 2.x
+
+## Description
+This is an Apache Maven Archetype to create a lambda function template using [AWS Java SDK 2.x][aws-java-sdk-v2]. The generated template
+has the optimized configurations and follows the best practices to reduce start up time.
+
+## Usage
+
+You can use `mvn archetype:generate` to generate a project using this archetype. See [maven archetype usage guidance][maven-archetype-usage] for more information.
+
+- Interactive mode
+
+```
+mvn archetype:generate \
+  -DarchetypeGroupId=software.amazon.awssdk \
+  -DarchetypeArtifactId=archetype-lambda \
+  -DarchetypeVersion=2.x\
+```
+
+- Batch mode
+
+```
+mvn archetype:generate \
+    -DarchetypeGroupId=software.amazon.awssdk \
+    -DarchetypeArtifactId=archetype-lambda \
+    -DarchetypeVersion=2.x\
+    -DgroupId=com.test \
+    -DartifactId=sample-project \
+    -Dservice=s3  \
+    -Dregion=us-west-2 \
+    -DinteractiveMode=false \
+```
+
+### Parameters
+      
+Parameter Name | Default Value | Description
+---|---|---
+`service` (required) | n/a | Specifies the service client to be used in the lambda function, eg: s3, dynamodb. You can find available services [here][java-sdk-v2-services].
```

Comment:
Can they just specify the module name (s3) *and* client name (S3Client) or something, so that we don't need to maintain a mapping?

Can we also generate instructions for adding more services? Maybe in a comment or something?

Reply:
They'd need to know the exact sync/async client names and I feel like that's kind of against the purpose of quickly bootstrapping a project from a template. 

Yeah, I can add some comments to explain how to add more services in the code.

Reply:
Sure, we can always write a build-tool to generate them pretty easily in the future.

Could we do a quick bash script to generate them for the currently-launched services? That shouldn't be too hard, if we're just hunting for things that extend SdkClient in each service module.

Reply:
Yeah, generated `serviceMapping.vm `  file that contains currently-launched services. I'll add the script in another PR to make it update the file automatically after every release.

Reply:
What if we added a build script to do it *during* the release, instead of having to do it after the release?

Reply:
Yeah, I misspoke. We can update it during the release like how we update the changelog.

Reply:
I misspoke too, I meant: can we write a build script, that as part of a mvn clean install generates the serviceMapping.vm used in this archetype?

Reply:
Like, a maven plugin or build tool that does it as part of the build.

Reply:
Yeah, I thought about it, but it'll take a really long time to build this specific module because it technically depends on all services. Release is probably fine because it builds all modules anyway but for development, if you are making changes in this module,  you'll have to wait for a long time for it to build.

Reply:
On second thought, it'll probably be easier to maintain if we keep the script within this module and build the file during build time.